wordpress-mobile/WordPress-Android
# Stories: Only show stories for sites that support them

Fixes https://github.com/Automattic/stories-android/issues/417.

In addition to the check for the stories feature flag being enabled, this PR now checks that the site is able to support the stories feature. This means either:

1. A WordPress.com simple site, or
2. A Jetpack-connected site with version 8.8 or higher

8.8 is the first Jetpack version to include the story block. The block is flagged as a beta block, which requires some extra steps to actually enable it on the site. When we're ready to release the stories feature, we'll want to update the minimum version set in this PR - this would be the sequence of events:

1. The story block in Jetpack is switched from beta to production
2. A new Jetpack version is released with the change (probably Jetpack 9.0)
3. We update the minimum version set in this PR to 9.0
4. The next WPAndroid release after step 3 is the first one we could use for a public beta

(This is being tracked on the stories project board.)

I also fixed the description toasts shown when long-pressing the FABs in the main activity and the post list. In the main activity we weren't accounting for stories, and in the post list no toast was being shown at all when stories were enabled.

Also, the FAB tooltip in the post list was always mentioning stories whether the feature was available or not - I fixed this as well.

Only one reviewer is really needed, ccing you @develric in case I missed anything important since I had to restructure `WPMainActivityViewModel` a bit.

### To test:
There's no need to test actually pushing a story to the site in this PR - a follow-up PR will remove the gallery fallback logic and call for this kind of test. Any site that isn't Tug's test site will just upload a gallery still as of this PR.

Instead, check that, for each site type:
- The options presented in the main activity and post list bottom sheets make sense
- The toasts shown when long pressing the FABs are consistent with the FAB options
- The tooltips shown over the FABs in both screens are consistent with the FAB options

These are the site type cases.

1. Simple WP.com site -> Story option shown ✅
2. Jetpack-connected site, version 8.8+ -> Story option shown ✅
3. Jetpack-connected site, version < 8.8 -> Story option not shown 🙅‍♂️
4. Self-hosted site -> Story option not shown 🙅‍♂️
5. (Optional) Atomic on WordPress.com site -> Story option shown ✅ (this is essentially the same as case 2, Atomic sites for this purpose are Jetpack-connected sites kept on the latest stable Jetpack version)

There's another degree of freedom in the main activity, which is whether the user has the ability to edit pages on the site. If not, there's no bottom sheet in the main activity if stories are disabled or unsupported.

(This is what the unit tests are mostly checking.)

Also check that the option is never shown if the feature flag is disabled, by e.g. [setting this to false](https://github.com/wordpress-mobile/WordPress-Android/blob/a3a756330e9cbf71dc75048563ed958a02bffda4/WordPress/build.gradle#L77).

PR submission checklist:
